src/froll.c

```diff
@@ -405,28 +405,38 @@ void frollsumExact(double *x, uint64_t nx, ans_t *ans, int k, double fill, bool
   }
 }
 
-inline void windowmax(const double * restrict x, uint64_t o, int k, double * restrict w, uint64_t *iw) {
-  for (int i=0; i<k-1; i++) {
-    //Rprintf("windowmax iteration %d, offset %d, first x val %f, testing x[o+i-k+1] >= w[0]: x[%d-%d+1] >= w[0]: %f >= %f: %d\n", i, o, x[o], i, k, x[o+i-k+1], w[0], x[o+i-k+1] >= w[0]);
-    if (x[o+i-k+1] >= w[0]) { // what if that is never satisfied? test!
-      iw[0] = o+i-k+1;
-      w[0] = x[iw[0]];
+
+inline void wmax(const double * restrict x, uint64_t o, int k, double * restrict w, uint64_t *iw, bool narm) {
+  if (narm) {
+    for (int i=0; i<k; i++) {
+      //Rprintf("wmax iter %d, offset %d, first x val %f, testing x[o+i-k+1] >= w[0]: x[%d-%d+1] >= w[0]: %f >= %f: %d\n", i, o, x[o], i, k, x[o+i-k+1], w[0], x[o+i-k+1] >= w[0]);
+      const uint64_t ii = o+i-k+1;
+      if (x[ii] >= w[0]) { // this never true if all x NAs and narm=TRUE
+        iw[0] = ii;
+        w[0] = x[ii];
+      }
     }
-  }
-}
-static inline void windowmaxnarm(const double * restrict x, uint64_t o, int k, bool narm, int *nc, double * restrict w, uint64_t *iw) {
```

Comment:
static that was lost here, due to auto resolving conflicts, will be added in froll2025max8, where it was originally included